templates/Engine method templates/ComputeTemplate/ComputeTemplate.cs

```diff
@@ -21,34 +21,41 @@
  */
 
 using System;
-using System.Collections;
-using System.Collections.Generic;
 using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
+using System.Collections.Generic;
+using System.ComponentModel;
+using BH.oM.Reflection.Attributes;
 using BH.oM.Base;
-using BH.oM.Structure.SectionProperties;
-using BH.oM.Common.Materials;
 
-namespace BH.Adapter.$ext_safeprojectname$
+namespace BH.Engine.External.SoftwareName // Replace SoftwareName with the name of your Toolkit software.
 {
-    public partial class $ext_safeprojectname$Adapter
+    public static partial class Compute
     {
-
         /***************************************************/
-        /**** Private methods                           ****/
+        /**** Public Methods                            ****/
         /***************************************************/
+		// All public engine methods are reflected into the UIs.
+		// Compute methods should be only those methods that are computationally-intensive
+		// or do not fall in any other Engine method category.
 
-        //The List<string> in the methods below can be changed to a list of any type of identification more suitable for the toolkit
-        //If no ids are provided, the convention is to return all elements of the type
-
-        private List<ISectionProperty> ReadSectionProperties(List<string> ids = null)
+        [Description("")]
+		[Input("","")]
+        [Output("", "")]
+        public static void $safeitemname$()
         {
-            //Implement code for reading section properties
-            throw new NotImplementedException();
+			throw new NotImplementedException();
         }
 
         /***************************************************/
+		
+		/***************************************************/
+        /**** Private Methods                           ****/
+        /***************************************************/
+		// Include here any private method that the public one might need to reference.
+		// Private methods are not reflected in the UIs.
+		
+		
+		/***************************************************/
```

Comment:
little bit nervous we will end up with this in many files where not needed, but a tradeoff I guess

Reply:
Removed